app/infra/walletUpdateService/walletUpdateService.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+import { ipcRenderer } from 'electron';
+import { ipcConsts } from '/vars';
+import { listenerCleanup } from '/infra/utils';
+
+class WalletUpdateService {
+  static getWalletUpdateStatus() {
+    ipcRenderer.send(ipcConsts.GET_WALLET_UPDATE_STATUS);
+    return new Promise<string, Error>((resolve: Function, reject: Function) => {
+      ipcRenderer.once(ipcConsts.GET_WALLET_UPDATE_STATUS_SUCCESS, (event, xml) => {
+        listenerCleanup({ ipcRenderer, channels: [ipcConsts.GET_WALLET_UPDATE_STATUS_SUCCESS, ipcConsts.WALLET_UPDATE_ERROR] });
+        resolve(xml);
+      });
+      ipcRenderer.once(ipcConsts.WALLET_UPDATE_ERROR, (event, args) => {
```

Comment:
move error listener to separate function, the error is global and can be caused by some other reason